pkg/app2/client.go

```diff
@@ -16,28 +19,34 @@ type Client struct {
 	pk  cipher.PubKey
 	pid ProcID
 	rpc RPCClient
-	lm  *idManager // contains listeners associated with their IDs
-	cm  *idManager // contains connections associated with their IDs
+	lm  *idmanager.Manager // contains listeners associated with their IDs
+	cm  *idmanager.Manager // contains connections associated with their IDs
 }
 
 // NewClient creates a new `Client`. The `Client` needs to be provided with:
 // - log: logger instance
 // - localPK: The local public key of the parent skywire visor.
 // - pid: The procID assigned for the process that Client is being used by.
-// - rpc: RPC client to communicate with the server.
-func NewClient(log *logging.Logger, localPK cipher.PubKey, pid ProcID, rpc RPCClient) *Client {
+// - sockFile: unix socket file to connect to the app server.
+// - appKey: application key to authenticate within app server.
+func NewClient(log *logging.Logger, localPK cipher.PubKey, pid ProcID, sockFile, appKey string) (*Client, error) {
```

Comment:
Can we have a `ClientConfig` structure for these inputs excluding `log *logging.Logger`?

We also need an additional function:
```golang
func ClientConfigFromEnv() ClientConfig { /*TODO*/ }
```
That generates the `ClientConfig` from envs.

Alternative solutions welcome.

Reply:
Done